samples/nrf9160/serial_lte_modem/src/slm_at_host.c

```diff
@@ -59,19 +67,40 @@ static size_t at_buf_len;
 static struct k_work cmd_send_work;
 static const char termination[3] = { '\0', '\r', '\n' };
 
+static u8_t uart_rx_buf[UART_RX_LEN][1];
```

Comment:
Why multidimensional array?

Reply:
Serial Async RX API requires array of buffer (buffer request and buffer release mechanism). For the purpose of AT string, the size buffer is limited to be one char (as AT string is terminated by special char).

Reply:
It can be single dimensional array, just pass pointer to single elements.
Asynchronous API has timeout mechanism to handle cases where receiving length is unknown. Single byte reception is, to put it mildly, suboptimal.